Simplify and fix right prompt

olbrew opened the pull request:
The right prompt shows a Vi mode indicator.
This commit rewrites it with native fish-shell functions, fixes the coloring
issue seen before and adds a new check for the vi mode used from fish v2.3
onward so it is only shown when vi mode is enabled.

This is the follow up pull request as discussed in #19

olbrew commented:
@sn0cr can you merge this?
Worth noting that the newest stable 2.3.1 fixes an [issue](https://github.com/fish-shell/fish-shell/issues/3067) with the mode_prompt and therefore it now has to be manually disabled which can easily be done by adding the following to your `fish.config`:

```
# Remove builtin vi mode indicator in favour of custom theme
function fish_mode_prompt
end function
```

I've also added this caveat to the README.

derekstavis commented:
Hey @olbrew I will have a look at your pull request and merge if it works for me. Thanks for the ping.

derekstavis commented on README.md:
```diff
@@ -26,7 +27,6 @@ For Mac users, I highly recommend iTerm 2 + Solarized Dark
 `set --export VIRTUAL_ENV_DISABLE_PROMPT 1`
-* Indicate vi mode. (If you've set `fish_vi_mode` in your config and don't like the ugly left prompt indicator you can solve this by replacing it with `set -g fish_key_bindings fish_vi_key_bindings` and then removing the `if set -q __fish_vi_mode` check at the bottom of the `fish_right_prompt.fish`)
-
+* Indicate vi mode. You will probably want to disable the default vi mode prompt. Add to your [`init.fish`](https://github.com/oh-my-fish/oh-my-fish#dotfiles): `function fish_mode_prompt; end function`
```
The command should be `function fish_mode_prompt; end`. There's no `end function` 😛

olbrew replied:
Ow yeah, good catch.

derekstavis commented:
The best thing to do if you'd like the theme to automatically remove the left vi indicator is to define an empty `fish_mode_prompt` function in `fish_mode_prompt.fish` file.

olbrew commented:
@derekstavis That's indeed even better. I've added it to the right prompt. That seems better than creating a new file just for the purpose of defining an empty function. I've also squashed the commits to have a cleaner history.

sn0cr commented:
Sorry that I'm replying that late. 
Are you ready to merge?

olbrew commented:
Yep, should be ready to go. I use it myself and it works against the latest version 👍.

sn0cr commented:
Thank you very much @olbrew for your contribution and @derekstavis for the help.
